sympy/solvers/bivariate.py

```diff
@@ -209,7 +209,7 @@ def _solve_lambert(f, symbol, gens):
         lhs = expand_log(log(lhs))
         rhs = log(rhs)
 
-    lhs = factor(lhs, deep=True)
+    lhs = factor(lhs)
```

Comment:
Here deep=True will solve most of the problems . Only this test case is failing if deep=True 
```assert solveset_real(5*x - 1 + 3*exp(2 - 7*x), x) == FiniteSet(Rational(1, 5) + LambertW(-21*exp(Rational(3, 5))/5)/7)```
Just wanted to let you know.

Reply:
Yeah! `deep=True` needs to be there. Untill the issue is fixed, I wanted to see how many tests are affected ;-)
I have opened #16302, after it gets merged the problem will be solved here.